### PR TITLE
Always show floating dock-panes on macOS.

### PR DESCRIPTION
**Pull request by mdickinson**
On Linux and Windows, undocked ("floating") dock panes are always visible, even when focus is switched to a different application. On macOS / Qt, by default the dock panes (which are Qt "tool" windows) are hidden when focus is switched. This PR sets the appropriate attribute to make sure that the dock pane window is always visible, even when the application loses focus, to give a consistent Tasks application behaviour across platforms.

### CODE REVIEW

**Comment by mdickinson** [comment]
I've verified that the attribute is harmless on Windows.

**Comment by mdickinson** [comment]
I'm not seeing a reasonable way to write a regression test for this. I could certainly test that created dock panes have their `WA_MacAlwaysShowToolWindow` attribute set, but I'm not sure how useful such a test would be. @corranwebster Is it worth me adding such a unit test?

**Comment by corranwebster** [comment]
Yep - I don't see any way to validate this has the desired effect, but it might be worth a test to check that the property is set to the correct flag value on OS X just as a regression test.

File: pyface/ui/qt4/tasks/tests/test_dock_pane.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+class TestTask(Task):
+    id = "test_task"
+    name = u"Test task"
+
+    def _default_layout_default(self):
+        return TaskLayout(left=PaneItem("test_dock_pane", width=200))
+
+    def create_central_pane(self):
+        # self.central_pane = central_pane = EditorAreaPane()
```
Comment (mdickinson) [inline pyface/ui/qt4/tasks/tests/test_dock_pane.py]: Whoops; left-over debugging code. Will remove.

**Comment by mdickinson** [comment]
I tested behaviour on wx / macOS - there again, dock panes are hidden when a different application is activated.

Given that, I'm not sure whether this PR should be merged at all. Is it more important that Canopy Data behave like other macOS applications, or is it more important to present a consistent behaviour across platforms?

Related: Apple's [HIG says](https://developer.apple.com/design/human-interface-guidelines/macos/windows-and-views/panels/):
> When an app is inactive, all of its panels should be hidden.

But maybe it's wrong for the IPython prompt and the data pane to be presented as panels in the first place...

@rkern @corranwebster thoughts?

**Comment by corranwebster** [comment]
Aha - `QDockWidget::DockWidgetFloatable`.  So maybe that's the answer.

**Comment by mdickinson** [comment]
My experience is that people value the undockability - I've seen users of applications based on the tasks framework undocking the Python shell as a matter of preference.

**Review by corranwebster — APPROVED** [review]
Approving this - it looks like it was complete and behaviour was good and tested, but didn't get approved/merged for some reason.

File: pyface/ui/qt4/tasks/tests/test_dock_pane.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
+# Copyright (c) 2014-2019 by Enthought, Inc., Austin, TX
```
Comment (rahulporuri) [inline pyface/ui/qt4/tasks/tests/test_dock_pane.py:1]:
```suggestion
# Copyright (c) 2014-2021 by Enthought, Inc., Austin, TX
```

**Comment by rahulporuri** [comment]
@mdickinson @corranwebster I merged the master branch into this branch and I updated the copyright header. Merging now.

**Pull request merged**